frame/benchmarking/src/analysis.rs

```diff
@@ -31,13 +32,48 @@ pub struct Analysis {
 	pub model: Option<RegressionModel>,
 }
 
+#[derive(Clone, Copy)]
 pub enum BenchmarkSelector {
 	ExtrinsicTime,
 	StorageRootTime,
 	Reads,
 	Writes,
 }
 
+#[derive(Debug)]
+pub enum AnalysisChoice {
+	/// Use minimum squares regression for analyzing the benchmarking results.
+	MinSquares,
+	/// Use median slopes for analyzing the benchmarking results.
+	MedianSlopes,
+	/// Use the maximum values among all other analysis functions for the benchmarking results.
+	Max,
+}
+
+impl Default for AnalysisChoice {
+	fn default() -> Self {
+		AnalysisChoice::MinSquares
```

Comment:
why isn't max the default, if we are to be pessimistic?

Reply:
its not clear to me that we should use `max` for everything. In general, min squares is a solid choice for an algorithm, this PR just gives us more options when there are weird results like we say with `remark_with_event`